angular/projects/spark-angular/src/lib/components/sprk-masthead/directives/sprk-masthead-nav-bar/sprk-masthead-nav-bar.directive.ts

```diff
@@ -0,0 +1,32 @@
+import { Directive, Input, HostBinding, ElementRef } from '@angular/core';
+
+@Directive({
+  selector: '[sprkMastheadNavBar]',
+})
+export class SprkMastheadNavBarDirective {
+  /**
+   * @ignore
+   */
+  constructor(public ref: ElementRef) {}
+
+  /**
+   * The value supplied will be assigned to the `data-id` attribute on the
+   * component. This is intended to be used as a selector for automated
+   * tools. This value should be unique per page.
+   */
+  @HostBinding('attr.data-id')
+  @Input()
+  idString: string;
+
+  /**
+   * The value supplied will be assigned to the
+   * `data-analytics` attribute on the element.
+   * Intended for an outside
+   * library to capture data.
+   */
+  @HostBinding('attr.data-analytics')
+  @Input()
+  analyticsString: string;
+
+  @HostBinding('class.sprk-c-Masthead__nav-bar') mastheadNavBarClass = true;
```

Comment:
We haven't been putting additionalClasses on directives since the user would use the class attribute. What are your thoughts on that?

Reply:
I have been adding `additionalClasses` to directives to keep the API consistent across components/directives. Users could also use `data-id` and `data-analytics` instead of `idString` and `analyticsString` but we provide those Inputs for consistency.

Reply:
Resolved in chat. We are going to not include additionalClasses on directives for now because it could be confusing to users for them when deciding to use additionalClasses or the regular class attribute.